glib/src/commonMain/kotlin/io/github/cgr/glib/LazyList.kt

```diff
@@ -154,6 +156,27 @@ class LazyList(private val generator: Iterator<Any?>) : List<Any?> {
             return result
         }
     })
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is List<*>) return false
+        for (i in this.indices) {
+            if (this[i] != other[i]) return false
+        }
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = 1
+        for (element in this) {
+            result = 31 * result + (element?.hashCode() ?: 0)
```

Comment:
I'm slightly concerned about overflow here, but I'm also coming from C++ where signed integer overflow is UB.

Reply:
Overflow is a given in `hashCode`ing, this is the standard algorithm. Overflow barely affects the result. In Java/Kolin overflow is not UB, it just... overflows.